### ISSUE TITLE
Need help

### ISSUE THREAD

**Issue by Perlenjaeger** [tarwirdur/open-fixdry-nt1#1]
Hi Tarwirdur Turon,

I don't know how to contact you. I just try it that way. I also want to modify my fixdry. Worked trough your Redme. Solderd a 8s103F3 on the bord. I hope I didn't toted the microcontroler while soldering. I also solderd the pin headers for the original JTAG connection. Now I connected a Stlink/V2. I'm connecting the VCC and GND thats traight forward. But the SWIN I just found on the Sensor connector pin SDA. I have not connected the reset line. Couldn't find a convinient way to connect it.
But now when when I try to flash. my output is like this.

```
> ./stm8flash/stm8flash -c stlinkv2 -p stm8s103f3 -w ./open-fixdry-nt1/build/STM8S103/STM8S103.hex
Determine FLASH area
STLink: v2, JTAG: v29, SWIM: v7, VID: 8304, PID: 4837
Due to its file extension (or lack thereof), "./open-fixdry-nt1/build/STM8S103/STM8S103.hex" is considered as INTEL HEX format!
4016 bytes at 0x8000... SWIM error 0x05
```

I also can not get a connection with the STM32 ST-LINK utility.

Do you have a Idee whats going wrong? Do I have the connections right?

Thank you very much

**Comment by tarwirdur** [comment]
Hi.

There are two versions of pcb. Which one do you have?

**First (with 2x sm74hc5950):**
![90467e](https://github.com/user-attachments/assets/56a2a503-bbf9-42dc-bd15-7f0ed8b92146)

If you have this version, then you need to connect stlink like this: (blue is RST pin)
![c41056](https://github.com/user-attachments/assets/3e8139a9-fc26-4884-8bd1-d07a4e886f97)


**Second (with TM1639):**
![im](https://github.com/user-attachments/assets/c31d2685-8bc0-4feb-bd6c-297834eafb31)

This version is also supported, but it requires minor changes to the pcb. (cut 2 tracks, add 2 wires and 1 capacitor)
And the firmware for this version is in a separate branch: https://github.com/tarwirdur/open-fixdry-nt1/tree/tm1639

**Comment by Perlenjaeger** [comment]
Thank you very much for this awsomely quick reply.

Of course I have the second version with the tm1639.
Which tracks have to get cut? And what capacitor as to be added? I can't see them on the picture.
And where to connect the st link on this version?

I'm excited to test it tomorrow.

**Comment by tarwirdur** [comment]
![s-20241017-06-32-14](https://github.com/user-attachments/assets/d2199ba1-1983-4712-b42b-9619827de690)

The capacitor should be installed between the track being cut and the ground. It is VCAP capacitor (pin 8 of mcu). It should be between 470nF and 3.3uF according to the datasheet.

Two new wires are visible in the photo.

Connection scheme to the programmer:
![88f9cef0-6c98-46f4-9ce2-5fb1b1c014df](https://github.com/user-attachments/assets/5d1afece-f3fe-455f-b8a4-68052c743674)

For initial flashing RST should be connected near `cut1`. For next flashing this is not necessary.

Here HEX file for tm1639 version: https://github.com/tarwirdur/open-fixdry-nt1/releases/download/v0.5/STM8S103-TM1639.hex

**Comment by Perlenjaeger** [comment]
Thank you very much.
I had to wait for the capacitors to arrive by post. So today I was able to make all the hardware changes and everything worked out. I flashed your image and I can set the temperatur till 90°C very nice. Very good work from you, now it's a geate dryer.
![IMG_20241024_215859548_PORTRAIT](https://github.com/user-attachments/assets/235057e4-89a0-4a2d-9489-6a2d99c13bb9)

But if I compile the tm1639 branch by myself the hex file differs from yours and it's also not working.
In the config.h the define #define VER_TM1639 is set. so I haven't changed anything in the branch.
From STM8-SPL-SDCC I'm on the master branch.

Any clue?

Did you tryed higher temperatures then 90°C? I want to Isolate the dryer with EPS and try to push it further 120°C would be perfect for nylon.

**Comment by Perlenjaeger** [comment]
Found it. Yesterday I hadn't cleand the build dir.
what went wrong.
on master branch build the hex file
switched to tm1639 branch and build the hex without cleaning
therfore it seams there where still some old binary files used and not rebuild for the correct tm1639.hex
thank you again for your greate work and help for my build.

**Comment by Perlenjaeger** [comment]
changing the temperatur settings in your well structured code was super easy. now it will heat up till 120°C lets see if the plastic will withstand it.

**Comment by tarwirdur** [comment]
> So today I was able to make all the hardware changes and everything worked out. 

Glad to hear it!

> Any clue?

Мaybe you forgot to `make clean` after changing the branch?

I launched an automatic build on the tm1639 branch and the output firmware is no different from what I applied manually.

build: https://github.com/tarwirdur/open-fixdry-nt1/releases/tag/v0.5-tm1639
action log: https://github.com/tarwirdur/open-fixdry-nt1/actions/runs/11517348058/job/32061845568
action source: https://github.com/tarwirdur/open-fixdry-nt1/blob/v0.5-tm1639/.github/workflows/release.yml

> Did you tryed higher temperatures then 90°C? I want to Isolate the dryer with EPS and try to push it further 120°C would be perfect for nylon.

I see that I didn't write the readme in enough detail. I'll try to fix that in a while.

The temperature displayed on the screen by default, as in the stock firmware, is not equal to the actual temperature on the sensor.

The temperature distribution in the box is not constant and near the heater it is much higher than the sensor.

This firmware has an option with temperature interpolation (by default), where the displayed temperature is scaled approximately as in the stock firmware.

There is a mode where the actual temperature on the sensor is displayed directly.

To switch between modes, turn off the dryer, press "MODE" and, without releasing it, "POWER".

Three digits will be displayed. 1 - interpolation mode, 2 - timer end sound, 3 - button sound.

___

I haven't printed with nylon myself, but I've read that nylon was successfully printed from this dryer, even before modifications.
